### PR TITLE
Additionally compare chainId in URL and in provider for trade state updates

### LINKED ISSUE #1391: Issues with URL when wallet does not support several networks
Related to #1317 and #1384

Reported in https://github.com/cowprotocol/cowswap/pull/1384#issuecomment-1314990553 (cases 3-4)

3. Tokens in the link are not removed when I connect a wallet using WC, and it does not support other chains but Ethereum (example: TrustWallet): 
return back

4. When I change a network indicator in the URL (to 4, 5, 100) being connected to a wallet that does not support a network switch (supports only 1), it would be great to return the indicator to 1.

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #1391, #1717

`useSetupTradeState()` didn't check the case, when `chainId` was changed while changing wallet

  # To Test

1. Connect to Metamask on Gnosis chain
2. Click `Change wallet` in the Account modal
3. Connect via Walletconnect to a wallet that doesn't support Gnosis chain (Trust for example)
- [ ] AR: Wallet provider uses Mainnet, but the URL still contain Gnosis chain id
- [ ] ER: ChainId in the URL was changed corresponding to provider's chainId

### CODE REVIEW

**Review by elena-zh — APPROVED** [review]
Hey @shoom3301 , great job!
Just a nitpick: when I switch back from another wallet to MM, I see a request to change a network in MM. Not critical though, as I can cancel it.
switch back
Let me know please if we can do anything with it or we can leave it as it is for now. :)

Thanks!

File: src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts [hunk 1]
```diff
@@ -47,6 +48,10 @@ function shouldSkipUpdate(tradeStateFromUrl: TradeState, tradeStateFromStore: Tr
   return chainIdIsNotChanged && recipientIsNotChanged && inputCurrencyIsNotChanged && outputCurrencyIsNotChanged
 }
 
+function compareChainIds(aChainId: Nullish<number>, bChainId: Nullish<number>): boolean {
```
Comment (alfetopito) [inline src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts:51]: I suggest you follow the same pattern as the fn above `areCurrenciesTheSame` and name it: `areChainIdsTheSame`

Comment (shoom3301) [reply src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts:51]: Thank you!

**Comment by shoom3301** [comment]
>when I switch back from another wallet to MM, I see a request to change a network in MM

@elena-zh, It's ok, seems logical to me.

**Review by anxolin — APPROVED** [review]

**Pull request merged**